### DISCUSSION TITLE
Calculations of total concentration generated and at the boundaries

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#18040] [Q&A General] [answered]
Dear MOOSE Group,
I have a concentration diffusion model with a source term (please see the attached figure).
In the model, the concentration is generated in Block 1 and diffused to the boundaries.
I would like to calculate the total concentration [moles] generated in Block 1 and the concentration [moles] on the top, right, bottom, and left boundaries to confirm the conservation.
I am looking for examples for these calculations.
Please help me.
Thomas

[Diffusion_Model_with_Source.pdf](https://github.com/idaholab/moose/files/6625835/Diffusion_Model_with_Source.pdf)

**Comment by GiudGiud** [comment, marked as answer]
Hello

There are plenty of diffusion-source examples on the website. Let me know If you need me to find one for you 
You should be able to use postprocessors to calculate the concentration:
- SideAverageValue for the boundaries
- ElementAverageValue for block 1 https://mooseframework.inl.gov/source/postprocessors/ElementExtremeValue.html

Guillaume
